osu.Game/Graphics/Cursor/CursorTrail.cs

```diff
@@ -116,22 +118,24 @@ protected override bool OnMouseMove(InputState state)
             if (lastPosition == null)
             {
                 lastPosition = state.Mouse.NativeState.Position;
+                resampler.AddPosition(lastPosition.Value);
                 return base.OnMouseMove(state);
             }
 
             Vector2 pos1 = lastPosition.Value;
-            Vector2 pos2 = state.Mouse.NativeState.Position;
-
-            Vector2 diff = pos2 - pos1;
-            float distance = diff.Length;
-            Vector2 direction = diff / distance;
+            foreach (Vector2 pos2 in resampler.AddPosition(state.Mouse.NativeState.Position))

```

Comment:
I get that the current implementation only even returns one result, but this will break if that ever changes. I'd probably add something like `pos1 = pos2;` at the end of each `foreach`  iteration so it can potentially support a different return behaviour that is not zero or one points.